### ISSUE TITLE
Symfony4: Route not found with path param with a dot

### ISSUE THREAD

**Issue by mrljaime** [symfony/symfony#25911]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | yes
| RFC?             | yes/no
| Symfony version  | 4.0.3

I have notice that `{_format}` placeholder avoid that routes matches with dots:

```php
/**
 * @Route("/report/{paysheet}")
 *
 * @param $paysheet
 * @return JsonResponse
*/
```

When I make a request with this path below `/report/5a29775c880fc0.79633949` I have this response:

```
The requested resource /report/5a29775c880fc0.79633949 was not found on this server.
```

Notice that path param is a `uniqid("", true)`

I even tryed to add in the requirements section a regular expression to catch that dot without any success.

**Comment by javiereguiluz** [comment]
Is `@Route("/report/{paysheet}")` your real and complete route annotation? Or are you using `@Route("/report/{paysheet}.{_format}")` ?

**Comment by mrljaime** [comment]
I did it just like my first comment, without `.{_format}` placeholder:

`@Route("/report/{paysheet}")`

**Comment by SamirAffani** [comment]
Do you have an other Route like /report/{something} in your controller ?

**Comment by mrljaime** [comment]
No, i just have a DefaultController class only with two methods and only one matches that route.

In 30 minutes I can have access yo my computer to create a Gist.

**Comment by SamirAffani** [comment]
Ok.

Copy your whole controller here.

Did you try to create a paysheet like 09Az (without '.') ? I never had an URL like that, maybe the problem come from that.

**Comment by mrljaime** [comment]
I did it to verify that dot was the problem, and without dot the route matches perfectly

**Comment by xabbuh** [comment]
@mrljaime if you are able to create a very small application that allows to reproduce the issue, that could help others to look into it to confirm whether that's a bug or if you can solve your issue in another way :)

**Comment by SamirAffani** [comment]
Generally, dots are a bad thing in URL.
Do you have to do that? Why do you need this dot?

**Comment by mrljaime** [comment]
I did this to avoid passing id’s like path param, instead I use uniqid(“”, true) to create an unique id to tracking rows at database but this one concat two strings just like that. 
Taken your comment, if is a bad practice I have to change all my database unique id’s, or use sumfony’s 3.x version that doesn’t has this problem. I have four microservices at production in that version without conflicts with the dots.

**Comment by SamirAffani** [comment]
Why you can't go on SF3 ?

**Comment by mrljaime** [comment]
My first approach was going over SF4 because flexibility and to be updated. I’m going to do my homework at SF3 but I like to think that is is really a bug and will be solve. I will trying to see the code and hope find the peace that broke that use case.

**Comment by xabbuh** [comment]
I was not able to reproduce your issue. Please create a (small) example application that allows to reproduce.

**Comment by xabbuh** [comment]
@mrljaime FYI, I have pushed my approach to https://github.com/xabbuh/issue-25911. You can base your changes on top of that.

**Comment by mrljaime** [comment]
Hello. 
This is exactly the project that I have with the problem.

https://github.com/mrljaime/symfony-issue-25911.git

**Comment by mrljaime** [comment]
I just clone your project @xabbuh . I ran a server and this is the result:

http://localhost:8000/report/5a60f8932e6864.14938346

`The requested resource /report/5a60f8932e6864.14938346 was not found on this server.`

Without dot:
http://localhost:8000/report/5a60f8932e6864

`{"foo":"bar"}`

**Comment by xabbuh** [comment]
Did you use PHP's built-in web server or did you set up Nginx/Apache/whatsoever?

**Comment by mrljaime** [comment]
I used PHP's buil-in web server:

You know: `php -S localhost:8000 -t public/`

I can try it with Apache or Nginx-FPM.

**Comment by xabbuh** [comment]
I used the built-in one too. Is it still the same when you use the `server:start` command instead?

**Comment by mrljaime** [comment]
Let me try that one. I'm configuring a virtual host over Apache to try it too.

**Comment by mrljaime** [comment]
@xabbuh 
I did it this server configuration over my Apache2

    ServerName syfmonyissue.io
    DocumentRoot /Users/jaime/Workspace/Osmos/aruane/public

    <Directory /Users/jaime/Workspace/Osmos/aruane/public>
        Options Indexes FollowSymLinks Includes ExecCGI
        AllowOverride All
        Require all granted

        <IfModule mod_rewrite.c>
            Options -MultiViews
            RewriteEngine On
            RewriteCond %{REQUEST_FILENAME} !-f
            RewriteRule ^(.*)$ index.php [QSA,L]
        </IfModule>

    </Directory>

    ErrorLog "/var/log/apache2/issue-25911_error.log"
    CustomLog "/var/log/apache2/issue-25911_access.log" common

That means that the problem doesn't look anymore:

http://symfonyissue.io/report/5a60f8932e6864.14938346
`{"foo":"bar"}`

**Comment by xabbuh** [comment]
Which PHP version did you use? And did you also try the `server:start` command?

**Comment by mrljaime** [comment]
I have PHP 7.1.8 
The problem doesn't replicate it with `server:start`

**Comment by xabbuh** [comment]
Can you confirm that `php -S localhost:8000 public/index.php` does work as well?

**Comment by mrljaime** [comment]
I confirm that `php -S localhost:8000 public/index.php` works perfectly.
Thanks.

**Comment by xabbuh** [comment]
Closing here then in favour of symfony/recipes#353. Thanks for confirming @mrljaime.

**Comment by Chumillas** [comment]
@mrljaime  Doing `php -S localhost:8000 public/index.php` doesn't load assets tho :(

**Comment by MatTheCat** [comment]
I think I'm impacted by this issue too; did someone understand why it occurs?

**Comment by Chumillas** [comment]
@MatTheCat  https://stackoverflow.com/questions/50053120/symfony4-route-not-found-with-path-param-if-contains-a-dot/50053199#50053199

**Comment by MatTheCat** [comment]
Thanks but there's still no “why”; plus I'm using nginx. I'll investigate monday.

**Comment by MatTheCat** [comment]
@nicolas-grekas seems my issue comes from https://github.com/symfony/symfony/commit/e16b3023c00d81beda6cf2cc31d44784054a51ec

WDYT?

**Comment by xabbuh** [comment]
@MatTheCat Could you create a new issue then please? The cause seems to be different and we should then track the issue until it is solved.

**Comment by MatTheCat** [comment]
@xabbuh my issue was in fact #27721 so it has been solved by #27736

**Comment by xabbuh** [comment]
@MatTheCat Thank you for checking and for the quick feedback! 🎉

**Comment by reevan100** [comment]
even I have the same issue in symfony 4.1. Production mode when symfony is configured to apache other than / other routes doesn't work but in development mode we are able to visit all routes.

**Comment by xabbuh** [comment]
@reevan100 That's probably related to missing rewrite rules in your web server configuration. https://symfony.com/doc/current/setup/web_server_configuration.html should contain some information about it.

**Comment by reevan100** [comment]
@mrljaime I think its showing that error for you its because you have not configured DirectoryIndex in your apache config to index.php. If you put index.php in front of your route it should work.

**Comment by mrljaime** [comment]
Hello @reevan100. 
I configured like I know and like symfony's web page says too in apache configuration. 
In both cases, the dot in the path has de problem, not with this response that I wrote before when open this issue:
`The requested resource /report/5a29775c880fc0.79633949 was not found on this server.`
In fact, that error was on dev environment because the dot was override the `index.php` default file. 
After put symfony's 4 under apache configuration, the error is founded when `._format` is defined, when you put `5a29775c880fc0.79633949` says that `79633949` is not a valid format.

Thanks anyway for your response.
Cheers.
